### PR TITLE
Databases week1

### PR DESCRIPTION
**Pull request by RodrigoGuinsbergPinto**
This is my database homework for Lesson1/week1.

### CODE REVIEW

**Review by dboyadzhiev — CHANGES_REQUESTED** [review]
pls, split your PR in two: one for DB and one for JS ohmeworks

**Comment by RodrigoGuinsbergPinto** [comment]
> pls, split your PR in two: one for DB and one for JS ohmeworks

I will try. Just don't have a clue about how to do this.

**Comment by dboyadzhiev** [comment]
> > pls, split your PR in two: one for DB and one for JS homework
> 
> I will try. Just don't have a clue about how to do this.

The easiest way, especially when you don't have experience with `git`, is to 
1. go to the branch `databases-week1`
2. copy your JS files (`javascript/javascript3/week1/`) somewhere, out of the project 
3. delete the folder (`javascript/javascript3/week1/`) from the project 
4. commit the changes
5. push the branch
6. done
7. go to the main branch 
8. create a new branch for `javascript3/week1/` 
9. copy back the files from `step 2` into the project
10. commit
11. push to the new JS branch

let me know if it helps

**Comment by RodrigoGuinsbergPinto** [comment]
Hi dboyadzhiev, I did what you suggested.
Hope it worked.

File: databases/week1/lesson1_homework.sql [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+-- 2. Find out how many tasks in the task table do not have a valid due date
+SELECT count(id) AS tasks_with_invalid_due_date
+FROM task
+WHERE due_date IS NULL;
+
+-- 3. Find all the tasks that are marked as done
+SELECT *
+FROM task
+WHERE status_id = 3;
```
Comment (dboyadzhiev) [inline databases/week1/lesson1_homework.sql]: in this case, you should use `JOIN` with the `status` table.  
We don't know what id `3` stand for

File: databases/week1/lesson1_homework.sql [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+-- 6. Get the single most recently created task
+SELECT *
+FROM task
+ORDER BY created DESC
+LIMIT 1;
+
+-- 7. Get the title and due date of all tasks where the title or description contains database
+SELECT title, due_date
+FROM task
+WHERE title REGEXP 'database' OR description REGEXP 'database';
```
Comment (dboyadzhiev) [inline databases/week1/lesson1_homework.sql]: The alternative is to use `like`
```suggestion
WHERE title like '%database%' OR description like '%database%';
```

File: databases/week1/lesson1_homework.sql [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+-- 8. Get the title and status (as text) of all tasks
+SELECT task.title, status.name AS status_description
+FROM task
+INNER JOIN status
```
Comment (dboyadzhiev) [inline databases/week1/lesson1_homework.sql]: you are missing the `ON` 
```suggestion
INNER JOIN status ON task.status_id = status.id
```

File: databases/week1/lesson1_homework.sql [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+ORDER BY status_description;
+
+-- 9. Get the name of each status, along with a count of how many tasks have that status
+SELECT status.name, COUNT(task.status_id) AS 'number_of_task'
```
Comment (dboyadzhiev) [inline databases/week1/lesson1_homework.sql]: the count should be by `task.id`
```suggestion
SELECT status.name, COUNT(task.id) AS 'number_of_task'
```

File: databases/week1/lesson1_homework.sql [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+FROM task
+INNER JOIN status
+ON task.status_id = status.id
+GROUP BY task.status_id;
```
Comment (dboyadzhiev) [inline databases/week1/lesson1_homework.sql]: The `group by` should be by `status.name`
```suggestion
GROUP BY status.name;
```

**Review by dboyadzhiev — CHANGES_REQUESTED** [review]

**Comment by RodrigoGuinsbergPinto** [comment]
Hi dboyadzhiev,
Thanks for the review.
Here is my file with suggested changes.

File: databases/week1/lesson1_homework.sql [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+-- 2. Find out how many tasks in the task table do not have a valid due date
+SELECT count(id) AS tasks_with_invalid_due_date
+FROM task
+WHERE due_date IS NULL;
+
+-- 3. Find all the tasks that are marked as done
+SELECT *
+FROM task
+INNER JOIN status ON task.status_id = status.id
+WHERE status_id = 3;
```
Comment (dboyadzhiev) [inline databases/week1/lesson1_homework.sql]: the ID of 'Done' can be different, so it is better to search by name instead
```suggestion
WHERE status.name = 'Done';
```

**Review by dboyadzhiev — APPROVED** [review]

**Comment by RodrigoGuinsbergPinto** [comment]
Hi Dimo, here is my last updated.
Thanks...

File: databases/week1/lesson1_homework.sql [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+SELECT *
+FROM task
+INNER JOIN status ON task.status_id = status.id
+WHERE status.name = 'Done';
+
+-- 4. Find all the tasks that are not marked as done
+SELECT *
+FROM task
+WHERE status_id != 3
+ORDER BY status_id;
```
Comment (dboyadzhiev) [inline databases/week1/lesson1_homework.sql]: I didn't say it explicitly but this query should be looking like the one above :)

**Review by dboyadzhiev — CHANGES_REQUESTED** [review]
1 more thing :)

**Comment by RodrigoGuinsbergPinto** [comment]
Hi Dimo,
Here is my file with changes to task 4.

**Review by dboyadzhiev — APPROVED** [review]

**Pull request merged**